src/packages/@ncigdc/utils/auth/awg.js

```diff
@@ -0,0 +1,137 @@
+import { throttle } from 'lodash';
+import urlJoin from 'url-join';
+import moment from 'moment';
+
+import consoleDebug from '@ncigdc/utils/consoleDebug';
+import {
+  AWG,
+  AWG_TOKEN_EXPIRY, // defaults to 5 minutes
+  AWG_TOKEN_RENEWAL_INTERVAL, // defaults to 0, disabled
+  FENCE,
+} from '@ncigdc/utils/constants';
+
+const createSessionRenewalTimestamp = now => {
+  consoleDebug(`Creating session timestamp at ${now}`);
+  localStorage.setItem('awg_sessionLastRenewed', now);
+  return !!localStorage.awg_sessionLastRenewed;
+};
+
+const removeSessionRenewalTimestamp = () => {
+  localStorage.removeItem('awg_sessionLastRenewed');
+  return !localStorage.awg_sessionLastRenewed;
+};
+
+const resetSessionRenewalTimestamp = now => (
+  removeSessionRenewalTimestamp() &&
+  createSessionRenewalTimestamp(now)
+);
+
+/* this is meant to avoid duplicate queries, because having multiple Relay Environments
+ * means renewal race conditions not worth dealing with at this point. ¯\_(ツ)_/¯
+ */
+let queryCache = null;
+const handleQueryCache = value => {
```

Comment:
Not particularly happy having to do this bit, but seems to get the job done to ensure no unnecessary duplicate calls from race conditions between the coexisting Relay environments.